### PR TITLE
Bump maxVersion for Palemoon 31.

### PR DESCRIPTION
**Pull request by mbecroft**
Tested working with github and gitlab. No guarantee of compatibility with other sites.

### CODE REVIEW

**Comment by mbecroft** [comment]
Gotcha. You are looking at the big picture. I have several web sites/SaaS products that have started to break in Pale Moon in the last 6 months. Not sure how long we can just keep patching individual sites. Anyway...I doubt palefill on 31 is a regression, and of you really want to know, you probably need a test suite involving dozens of web sites. Personally I would just release this and let someone file a bug if there is a regression. There are also larger issues around browser compatibility, especially webcomponents. It's all rather difficult for a small team to maintain. I hope I can make time to contribute a bit next few weeks, as I do love Pale Moon and wish I did not have to switcth to Chrome for an increasing number of web sites!

**Comment by mbecroft** [comment]
TL;DR everything looks fine re compatibility, so just release it for PM 31. Worst case, someone hits a blocking issue and has to temporarily regress, while we resolve the root cause.

**Comment by martok** [comment]
Implemented as https://github.com/martok/palefill/commit/e7a6817cb3a62999a2677a4ce757ead1d238e370.

> You are looking at the big picture. I have several web sites/SaaS products that have started to break in Pale Moon in the last 6 months. Not sure how long we can just keep patching individual sites.

Yeah, I know: perfect is the enemy of good. My ideal "big picture" future would be one where I can mark this repo as archived because it's not needed anymore...

> There are also larger issues around browser compatibility, especially webcomponents and new JS syntax changes. It's all rather difficult for a small team to maintain.

I'm very glad the UXP development repo has been opened again, it seems like there are many more contributions than before. Hopefully this is something more sustainable this time. A lot of WC is in by this point, I'd guess this is the feature that gets finalised first. JS syntax is "mostly ok" with `??` and `?.` implemented, the next major thing is the regex engine. That looks like a nightmare.
Mostly everything else can be polyfilled, we can do that even for the moving target it is.
